python/ray/_private/resource_spec.py

```diff
@@ -198,6 +200,40 @@ def resolve(self, is_head: bool, node_ip_address: Optional[str] = None):
         except Exception:
             logger.exception("Could not parse gpu information.")
 
+        # AWS NeuronCore detection and configuration
```

Comment:
let's create a helper function for this block. we can move neuron related code into python/ray/_private/accelerator.py